apis/placement/v1alpha1/rebalance_types.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:scope=Cluster,categories={fleet,fleet-placement}
+// +kubebuilder:subresource:status
+// +kubebuilder:storageversion
+
+// ClusterResourcePlacementRebalance is a rebalancing attempt on a ClusterResourcePlacement
+// object; one may use this API to ask Fleet to produce a new set of scheduling decisions
+// for the ClusterResourcePlacement object. This is most helpful in cases where certain
+// clusters in the fleet have become hotspots and the developers would like move select
+// workloads to less resource strained clusters to improve the system utilization.
+//
+// The execution of the rebalacing operation is subject to the rollout strategy defined
+// in the ClusterResourcePlacement object. This behavior is akin to editing the placement policy
+// of the ClusterResourcePlacement object; disruption budget would not apply to this operation.
```

Comment:
"disruption budget would not apply to this operation" Why not? Is that because customers don't want disruption budget to be enforced, or we can't achieve it?

Reply:
Disruption budget applies to all voluntary disruptions including "updating a deployment's pod template causing a restart", except for a few like deleting a deployment or a pod directly.

I thing this rebalance API belongs to voluntary disruptions.

Could you elaborate more on the k8s behavior you have in mind?